corpus_test.go

```diff
@@ -0,0 +1,185 @@
+package cedar
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"slices"
+	"strings"
+	"testing"
+)
+
+// jsonEntity is not part of entityValue as I can find
+// no evidence this is part of the JSON spec.  It also
+// requires creating a parser, so it's quite expensive.
+type jsonEntity EntityUID
+
+func (e *jsonEntity) UnmarshalJSON(b []byte) error {
+	if string(b) == "null" {
+		return nil
+	}
+	var input EntityUID
+	if err := json.Unmarshal(b, &input); err != nil {
+		return err
+	}
+	*e = jsonEntity(input)
+	return nil
+}
+
+type corpusTest struct {
+	Schema         string `json:"schema"`
+	Policies       string `json:"policies"`
+	ShouldValidate bool   `json:"shouldValidate"`
+	Entities       string `json:"entities"`
+	Requests       []struct {
+		Desc      string     `json:"description"`
+		Principal jsonEntity `json:"principal"`
+		Action    jsonEntity `json:"action"`
+		Resource  jsonEntity `json:"resource"`
+		Context   Record     `json:"context"`
+		Decision  string     `json:"decision"`
+		Reasons   []string   `json:"reason"`
+		Errors    []string   `json:"errors"`
+	} `json:"requests"`
+}
+
+type FileDataPointer struct {
+	Position int64
+	Size     int64
+}
+
+type FileDataMap struct {
+	buf   io.ReaderAt
+	files map[string]FileDataPointer
+}
+
+func NewFileDataMap(buf io.ReaderAt) FileDataMap {
+	return FileDataMap{
+		buf:   buf,
+		files: make(map[string]FileDataPointer),
+	}
+}
+
+func (fdm *FileDataMap) AddFileData(path string, position int64, size int64) {
+	fdm.files[path] = FileDataPointer{position, size}
+}
+
+func (fdm FileDataMap) GetFileData(path string) ([]byte, error) {
+	fdp := fdm.files[path]
+	content := make([]byte, fdp.Size)
+	_, err := fdm.buf.ReadAt(content, fdp.Position)
+	if err != nil {
+		return nil, err
+	}
+
+	return content, nil
+}
+
+func TestCorpus(t *testing.T) {
+	t.Parallel()
+
+	corpusArchive, err := os.Open("corpus-tests.tar.gz")
+	if err != nil {
+		t.Fatal("error opening corpus archive", err)
+	}
+	defer corpusArchive.Close()
+
+	gzipReader, err := gzip.NewReader(corpusArchive)
+	if err != nil {
+		t.Fatal("error reading corpus compressed archive header", err)
+	}
+	defer gzipReader.Close()
+
+	buf, err := io.ReadAll(gzipReader)
```

Comment:
`buf` ends up being about 14MB in size with the current version of the integration tests. Not great, not terrible.